R/seqCluster.R

```diff
@@ -0,0 +1,114 @@
+#' Function for asignment clusters based on sequencies similarity
+#'
+#' @importFrom purrr map map_lgl map2 map2_chr map_df map2_lgl
+#' @importFrom magrittr %>% %<>%
+#' @importFrom reshape2 melt
+#' @importFrom dplyr group_by mutate ungroup select cur_group_id left_join
+#' @importFrom igraph graph_from_data_frame clusters
+#' @importFrom reshape2 melt
+#' @importFrom tibble rownames_to_column
+#'
+#' @description Graph clustering based on distances between sequences
+#'
+#' @usage
+#'
+#' seqCluster(.data, .dist, .perc_similarity, .nt_similarity, .fixed_threshold)
+#'
+#' @param .data The data which was used to caluculate .dist object. Can be \link{data.frame},
+#' \link{data.table}, or a list of these objects.
+#'
+#' Every object must have columns in the immunarch compatible format \link{immunarch_data_format}
+#'
+#' @param .dist List of distance objects produced with \link{seqDist} function.
+#'
+#' @param .perc_similarity   Numeric value between 0-1  specifying the maximum acceptable weight of an edge in a graph.
+#'                           This threshold depends on the length of sequences.
+#' @param .nt_similarity     Numeric between 0-sequence length specifying
+#'                           the threshold of allowing a 1 in n nucleotides mismatch in sequencies.
+#' @param .fixed_threshold     Numeric specifying the threshold on the maximum weight of an edge in a graph.
+#'
+#' @return
+#'
+#' Immdata data format object. Same as .data, but with extra 'Cluster' column with clusters assigned.
+#'
+#' @examples
+#'
+#' data(immdata)
+#' # In this example, we will use only 2 samples for time saving
+#'
+#' dist_result <- seqDist(immdata$data[1:2])
+#' cluster_result <- seqCluster(immdata$data[1:2], dist_result, .fixed_threshold = 1)
```

Comment:
Seems like a perfect place for bugs to thrive. The interface forces user to remember what data was used to obtain `dist_result` and pass the same subset (`immdata$data[1:2]`) in `seqCluster` along with `dist_result` object